rust/src/context.rs

```diff
@@ -529,3 +526,262 @@ pub fn add_to_linker<T: 'static>(
+    //!
+    //! // ...Then add it to the linker.
+    //! dylibso_observe_sdk::context::component::add_to_linker(&mut linker)?;
+    //!
+    //! # let (cmd, _) = wasmtime_wasi::preview2::command::Command::instantiate_async(&mut store, &component, &linker).await?;
+    //! # let run = cmd.wasi_cli_run();
+    //! // Once you're done with wasm, call `shutdown()`, which shuts down the collector associated
+    //! // with the Observe SDK.
+    //! let state = store.into_data();
+    //! state.sdk.shutdown().await?;
+    //! # Ok(())
+    //! # }
+    //! ```
+    //!
+    //! To see an example integrating both Wasi preview 2 and the Observe SDK, see
+    //! `rust/examples/otel-stdout-components.rs` in the [observe sdk
+    //! repo](https://github.com/dylibso/observe-sdk/).
+    use crate::adapter::TraceContext;
+
+    use super::*;
+    use wasmtime::component::Linker;
+
+    /// Provide access to [`ObserveSdk`] from a [`wasmtime::Store`]'s inner object.
+    ///
+    /// ```rust
+    /// use dylibso_observe_sdk::context::component::{ ObserveSdk, ObserveSdkView };
+    ///
+    /// struct MyState {
+    ///   sdk: ObserveSdk,
+    /// }
+    ///
+    /// impl ObserveSdkView for MyState {
+    ///     fn sdk_mut(&mut self) -> &mut ObserveSdk {
+    ///         &mut self.sdk
+    ///     }
+    /// }
+    /// ```
+    pub trait ObserveSdkView {
+        fn sdk_mut(&mut self) -> &mut ObserveSdk;
+    }
+
+    // Hide the bindgen-generated modules from rustdoc by using an "internal" module.
+    mod internal {
+        wasmtime::component::bindgen!({
+            interfaces: r#"
+                import dylibso:observe/api;
+                import dylibso:observe/instrument;
+            "#,
+            path: "../wit",
+            async: false
+        });
+    }
+
+    use internal::dylibso::observe::api::{ MetricFormat, LogLevel, Host as ApiHost };
+    use internal::dylibso::observe::instrument::Host as InstrumentHost;
+
+    /// A data structure backing ObserveSdk bindings: contains private information mapping
+    /// function ids to names and contexts for collector channels.
+    ///
+    /// Instantiate by calling
+    /// [`AdapterHandle::build_observe_sdk`](crate::adapter::AdapterHandle::build_observe_sdk)
+    /// using any adapter type:
+    ///
+    /// ```no_run
+    /// # use dylibso_observe_sdk::context::component::ObserveSdk;
+    /// use dylibso_observe_sdk::adapter::zipkin::ZipkinAdapter;
+    /// # fn main() -> anyhow::Result<()> {
+    /// let args: Vec<_> = std::env::args().skip(1).collect();
+    /// let wasm_data = std::fs::read(&args[0])?;
+    /// let zipkin = ZipkinAdapter::create();
+    ///
+    /// // NB: the ": ObserveSdk" type annotation here isn't necessary, it's just to indicate the
+    /// // return type.
+    /// let observe_sdk: ObserveSdk = zipkin.build_observe_sdk(&wasm_data,
+    /// Default::default()).unwrap();
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub struct ObserveSdk {
```

Comment:
I don't feel strongly about this, just wondering if `ObserveCtx` would be a better name. That would imply these renames as well:
- `ObserveSdkView` ->  `ObserveCtxView` 
- `build_observe_sdk` -> `build_observe_ctx`